components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/datasource/QueryExecutor.java

```diff
@@ -66,17 +62,18 @@ public boolean executeInsertQuery(String uuid, byte[] businessRule, int deployme
         PreparedStatement statement = null;
         try {
             conn = dataSource.getConnection();
-            conn.setAutoCommit(true);
+            conn.setAutoCommit(false);
```

Comment:
You don't need to set auto commit here, a user can change the commit level in Datasource configuration as below:

e.g:
configuration:
        isAutoCommit: false